File: frontend/src/types/recommendation.ts
// Mirrors backend/app/schemas/recommendation.py exactly -- see
// docs/features/74-content-learning-loop.md.

export interface Recommendation {
  dimension: "pillar" | "format" | "hook" | "emotion";
  label: string;

  weight: number;
  historical_performance: number;
  sample_confidence: number;
  recency_factor: number;

  confidence_tier: "insufficient" | "low" | "medium" | "high";
  trend: "rising" | "stable" | "underperforming" | "insufficient_data";
  sample_size: number;
  linked_sample_size: number;

  reasons: string[];
}

File: backend/app/schemas/recommendation.py
from pydantic import BaseModel, ConfigDict


class RecommendationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    dimension: str
    label: str

    weight: float
    historical_performance: float
    sample_confidence: float
    recency_factor: float

    confidence_tier: str
    trend: str
    sample_size: int
    linked_sample_size: int

    reasons: list[str]
